components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx

```diff
@@ -117,14 +142,29 @@ const EditVisibleAssetsModal = (props: Props) => {
   }
 
   const tokenList = React.useMemo(() => {
-    const visibleContracts = userVisibleTokensInfo.map((token) => token.contractAddress)
-    const fullList = visibleContracts.includes('') ? fullAssetList : [nativeAsset, ...fullAssetList]
-    const notVisibleList = fullList.filter((token) => !visibleContracts.includes(token.contractAddress))
-    return selectedNetwork.chainId !== BraveWallet.MAINNET_CHAIN_ID
-      ? visibleContracts.includes('')
-        ? userVisibleTokensInfo
-        : [...userVisibleTokensInfo, nativeAsset]
-      : [...userVisibleTokensInfo, ...notVisibleList]
+    const visibleContracts = isUserVisibleTokensInfoEmpty
+      ? []
+      : userVisibleTokensInfo.map((token) => token.contractAddress)
+
+    const fullList = visibleContracts.includes('')
+      ? fullAssetList
+      : [nativeAsset, ...fullAssetList]
+
+    const filteredTokenRegistry = fullList.filter((token) => !visibleContracts.includes(token.contractAddress))
+
+    // If selected network is not Ethereum Mainnet we dont return
+    // any tokens from the ERCTokenRegistry
+    if (selectedNetwork.chainId !== BraveWallet.MAINNET_CHAIN_ID) {
+      return isUserVisibleTokensInfoEmpty
+        ? [nativeAsset]
+        : visibleContracts.includes('')
+          ? userVisibleTokensInfo
+          : [...userVisibleTokensInfo, nativeAsset]
+    } else {
+      return isUserVisibleTokensInfoEmpty
+        ? filteredTokenRegistry
+        : [...userVisibleTokensInfo, ...filteredTokenRegistry]
+    }
   }, [fullAssetList, userVisibleTokensInfo, selectedNetwork])
```

Comment:
This block will require a rebase on top of https://github.com/brave/brave-core/pull/11881 stuff. Basically, we don't treat `MAINNET_CHAIN_ID` specially anymore.

Reply:
Will rebase once yours has landed!

Reply:
Tested and works good with #11881 
https://user-images.githubusercontent.com/40611140/150204682-c2ffd1b6-b502-4f48-9c04-951fea5e2f4a.mov